### PR TITLE
Add ability to use slim in conjunction with ejs and jst in sprockets 4

### PR DESCRIPTION
**Pull request by stiller-leser**
We use slim in our rails app in conjunction with ejs and jst (file ending: *.js.jst.ejs.slim). Sprockets 4 won't recognize slim files as 'application/javascript' and hence fail when used in the above combination. I am aware that this is an edge case, but it prevents us from switching to sprockets 4.

### CODE REVIEW

File: lib/slim-rails/register_engine.rb [hunk 1]
```diff
@@ -30,6 +30,7 @@ def self._register_engine(config)
         config.assets.configure do |env|
           if env.respond_to?(:register_transformer) && Sprockets::VERSION.to_i > 3
             env.register_mime_type 'text/slim', extensions: ['.slim', '.slim.html']#, charset: :html
+            env.register_mime_type 'application/javascript', extensions: ['.js.jst.ejs.slim']
```
Comment (januszm) [inline lib/slim-rails/register_engine.rb]: This seems to be very specific and tailored for your use case, but I think that if we're going add this, then let's do it in a more general form, how about:

```ruby
env.register_mime_type 'application/javascript', extensions: %w[.js.slim .jst.slim .ejs.slim]
```

**Comment by januszm** [comment]
Could you also add test similar to the one we already have for html?

```ruby
      asset_path = File.join(dir, 'app', 'assets', 'html', 'test.slim')
      FileUtils.mkdir_p(File.dirname(asset_path))
      File.write(asset_path, ".test\n  | hi")
```

of course let's add some JS instead of html

File: lib/slim-rails/register_engine.rb (hunk 1 above)
Comment (stiller-leser) [reply lib/slim-rails/register_engine.rb]: Happy to do so. However I did realize that it is not working as intended. Will continue investigation and report back once solved.

**Comment by januszm** [comment]
@stiller-leser I know it's been a while, are you still interested in finishing this PR or should we rather fork this and continue on new branch?

**Comment by stiller-leser** [comment]
Hi @januszm,

what a blast from the past. I have since completly changed job profiles and companies as well. So I do not even know anymore why / where this was needed. You are probably better of forking this PR.

Sorry, I can't continue with this. If there is anything I can do, let me know and I will try to help out.

Best regards,
stiller-leser

**Pull request closed without merging**